common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java

```diff
@@ -0,0 +1,626 @@
+        }
+
+        @Override
+        String getNotification(Matcher matcher) {
+            return ChatFormatting.AQUA + "+3 minutes" + ChatFormatting.GRAY + " speed boost";
+        }
+    }
+
+    private class ToolDurabilityRedirector extends Redirector {
+        private static final Pattern UNCOLORED_SYSTEM_PATTERN = Pattern.compile(
+                "^Your tool has 0 durability left! You will not receive any new resources until you repair it at a Blacksmith.$");
+
+        @Override
+        Pattern getUncoloredSystemPattern() {
+            return UNCOLORED_SYSTEM_PATTERN;
+        }
+
+        @Override
+        RedirectAction getAction() {
+            return toolDurability;
+        }
+
+        @Override
+        String getNotification(Matcher matcher) {
+            return ChatFormatting.DARK_RED + "Your tool has 0 durability!";
+        }
+    }
+
+    private class UnusedAbilityPointsRedirector extends Redirector {
+        private static final Pattern UNCOLORED_SYSTEM_PATTERN = Pattern.compile(
+                "You have (\\d+) unused Ability Points?! Right-Click while holding your compass to use them");
+
+        @Override
+        Pattern getUncoloredSystemPattern() {
+            return UNCOLORED_SYSTEM_PATTERN;
+        }
+
+        @Override
+        RedirectAction getAction() {
+            return unusedPoints;
+        }
+
+        @Override
+        String getNotification(Matcher matcher) {
+            String unusedAbilityPoints = matcher.group(1);
+
+            return getUnusedAbilityPointsMessage(unusedAbilityPoints);
+        }
+
+        public static String getUnusedAbilityPointsMessage(String unusedAbilityPoints) {
+            return ChatFormatting.DARK_AQUA + "You have " + ChatFormatting.BOLD + unusedAbilityPoints
+                    + ChatFormatting.RESET + ChatFormatting.DARK_AQUA + " unused ability points";
+        }
+    }
+
+    private class UnusedSkillAndAbilityPointsRedirector extends Redirector {
+        private static final Pattern UNCOLORED_SYSTEM_PATTERN = Pattern.compile(
+                "You have (\\d+) unused Skill Points? and (\\d+) unused Ability Points?! Right-Click while holding your compass to use them");
+
+        @Override
+        Pattern getUncoloredSystemPattern() {
+            return UNCOLORED_SYSTEM_PATTERN;
+        }
+
+        @Override
+        RedirectAction getAction() {
+            return unusedPoints;
+        }
+
+        @Override
+        List<String> getNotifications(Matcher matcher) {
+            String unusedSkillPoints = matcher.group(1);
+            String unusedAbilityPoints = matcher.group(2);
+
+            return List.of(
+                    UnusedSkillPointsRedirector.getUnusedSkillPointsMessage(unusedSkillPoints),
+                    UnusedAbilityPointsRedirector.getUnusedAbilityPointsMessage(unusedAbilityPoints));
+        }
+
+        @Override
+        String getNotification(Matcher matcher) {
+            // We still need to implement this
```

Comment:
At first sight, this got me really confused. I thought, why did we not just implement this one notification? Then, it hit me. This is an abstract method and you mean that we need to impl this even though it won't be used!

```suggestion
            // We still need to implement this method, but it should not be called
```

Reply:
What I really wanted to express in the top-level abstract class that you need to implement *either* getNotification, if you have just a single notification, which is by far the most common case, *or* getNotifications, if you have multiple. 

I could have solved it by having a default implementation for getNotification which just returned null, but that would mean that we would not get the compiler screaming if we do wrong when adding more "normal" redirectors.

Or I could have solved it by always returning a List<String> (basically, only keep getNotifications()) but that would have made all "normal" implementations more complicated.

Hm. I should of course make a interface and a common base class for the simple notifications. 🤦 Sometimes a night's sleep make wonders with your ability to see your code.